packages/grid/x-data-grid-pro/src/internals/hooks/features/export/serializer/excelSerializer.ts

```diff
@@ -0,0 +1,181 @@
+        const formattedValueOptions = getFormattedValueOptions(
+          cellParams.colDef,
+          valueOptions,
+          api,
+        );
+        dataValidation[column.field] = {
+          type: 'list',
+          allowBlank: true,
+          formulae: [formattedValueOptions.map((x) => `"${x}"`)],
+        };
+        const formattedValue = getCellParams(id, column.field).formattedValue;
+        if (process.env.NODE_ENV !== 'production') {
+          if (String(cellParams.formattedValue) === '[object Object]') {
+            warnInvalidFormattedValue();
+          }
+        }
+        row[column.field] =
+          typeof formattedValue === 'object' ? formattedValue.label : formattedValue;
+        break;
+      }
+      case 'boolean':
+      case 'number':
+      case 'date':
+      case 'dateTime':
+        row[column.field] = getCellParams(id, column.field).value;
+        break;
+      case 'actions':
+        break;
+      default:
+        row[column.field] = getCellParams(id, column.field).formattedValue;
+        if (process.env.NODE_ENV !== 'production') {
+          if (String(cellParams.formattedValue) === '[object Object]') {
+            warnInvalidFormattedValue();
+          }
+        }
+        break;
+    }
+  });
+  return {
+    row,
+    dataValidation,
+    outlineLevel,
+  };
+};
+
+const serialiseColumn = (column: GridColDef) => {
+  const { field } = column;
+
+  return {
+    key: field,
+    // Excel width must stay between 0 and 255 (https://support.microsoft.com/en-us/office/change-the-column-width-and-row-height-72f5e3cc-994d-43e8-ae58-9774a0905f46)
+    // From the example of column width behavior (https://docs.microsoft.com/en-US/office/troubleshoot/excel/determine-column-widths#example-of-column-width-behavior)
+    // a value of 10 corresponds to 75px. This is an approximation, because column width depends on the the font-size
+    width: Math.min(255, column.width ? column.width / 7.5 : 8.43),
+  };
+};
+
+interface BuildExcelOptions {
+  columns: GridStateColDef[];
+  rowIds: GridRowId[];
+  getCellParams: (id: GridRowId, field: string) => GridCellParams;
+  includeHeaders: boolean;
+  exceljsPreprocess?: (processInput: GridExceljsProcessInput) => Promise<void>;
+  exceljsPostprocess?: (processInput: GridExceljsProcessInput) => Promise<void>;
+}
+
+export async function buildExcel(options: BuildExcelOptions, api): Promise<Excel.Workbook> {
+  const { columns, rowIds, getCellParams, includeHeaders, exceljsPreprocess, exceljsPostprocess } =
+    options;
+
+  const excelJS = await getExcelJs();
+  const workbook: Excel.Workbook = new excelJS.Workbook();
+  const worksheet = workbook.addWorksheet('Sheet1');
+
+  worksheet.columns = columns.map((column) => serialiseColumn(column));
+
+  if (exceljsPreprocess) {
+    await exceljsPreprocess({
+      workbook,
+      worksheet,
+    });
+  }
+
+  if (includeHeaders) {
+    worksheet.addRow(columns.map((column) => column.headerName || column.field));
```

Comment:
Not for this PR
But it make me more confident that we should do https://github.com/mui/mui-x/pull/3959#issuecomment-1042699805